kafka/pykafka/cluster.py

```diff
@@ -96,11 +102,19 @@ def _update_topics(self, metadata):
             self._topics.pop(name)
         # Add/update partition information
         for name, meta in metadata.iteritems():
-            if name not in self._topics:
-                self._topics[name] = Topic(self._brokers, meta)
-                logger.info('Adding topic %s', self._topics[name])
-            else:
-                self._topics[name].update(meta)
+            if not self._should_exclude_topic(name):
+                if name not in self._topics:
+                    self._topics[name] = Topic(self, meta)
+                    logger.info('Adding topic %s', self._topics[name])
```

Comment:
Not your changes, but I just noticed that this really should be "Discovered topic" instead of "Adding".  We're adding it to the internal model, but it was in Kafka before we got here.